app/src/main/java/com/example/habitsmasher/ui/history/HabitEventListFragment.java

```diff
@@ -0,0 +1,294 @@
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.Query;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+import com.google.firebase.storage.UploadTask;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.UUID;
+
+/**
+ * HabitEventListFragment class
+ * Responsible for data and UI handling of the habit event list
+ */
+public class HabitEventListFragment extends Fragment {
+    // Initialize variables
+    private static final String TAG = "HabitEventListFragment";
+
+    private HabitEventItemAdapter _habitEventItemAdapter;
+    private Habit _parentHabit;
+    private User _user;
+    private HabitEventList _habitEventList;
+
+    FirebaseFirestore _db = FirebaseFirestore.getInstance();
+
+    /**
+     * Default constructor
+     * @param parentHabit The habit for which the habit event list is
+     * @param parentUser The user for which the habit event list is
+     */
+    public HabitEventListFragment (Habit parentHabit, User parentUser) {
+        super();
+        this._parentHabit = parentHabit;
+        this._user = parentUser;
+    }
+
+    /**
+     * Use this factory method to create a new instance of
+     * this fragment using the provided parameters.
+     *
+     * @param parentHabit (Habit): The habit for which the habit events are being displayed
+     * @return A new instance of fragment HabitEventListFragment.
+     */
+    public static HabitEventListFragment newInstance(Habit parentHabit, User parentUser) {
+        HabitEventListFragment fragment = new HabitEventListFragment(parentHabit, parentUser);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        // Get context and query
+        Context context = getContext();
+        Query query = getListOfHabitEventsFromFirebase(_user.getUsername());
+
+        try {
+            // Populate the list with existing items in the database
+            FirestoreRecyclerOptions<HabitEvent> options = new FirestoreRecyclerOptions.Builder<HabitEvent>()
+                    .setQuery(query, HabitEvent.class)
+                    .build();
+
+            // Set item adapter and habit event list
+            _habitEventItemAdapter = new HabitEventItemAdapter(options);
+            _habitEventList = _parentHabit.getHabitEvents();
+        }
+        catch (Error e){
+            // Try catch statement is needed so code doesn't break if there's no events yet, and thus no possible query
```

Comment:
Let's at least add a **log message** here so the body isn't empty, allowing us to step through the code while debugging in the future.